ank/doc/swdesign/README.md

```diff
@@ -506,6 +506,265 @@ Needs:
+| **restart** | It specifies whether the workload shall be restarted when it exits | One of: `true`,`false` | true |
+| **updateStrategy** | It specifies the update strategy. | One of: `UNSPECIFIED`, `AT_LEAST_ONCE`, `AT_MOST_ONCE` | true |
+| **accessRights** | It specifies lists of access rules fpr `allow` and `deny`. | Not fully specified yet and shall be set to empty list for both. | true |
+| **tags** | It specifies a list use defined key/value objects. | A list of { `key`: some_string, `value`: some_string} or empty| true |
+| **runtimeConfig** | It specifies the configuration for the runtime whose configuration structure is specific for each runtime as a _string_ | As a _string_ from one of: [PodmanRuntimeConfig](#podmanruntimeconfig), [PodmanKubeRuntimeConfig](#podmankuberuntimeconfig),  | true |
+
+###### PodmanRuntimeConfig
+
+The runtime configuration for the podman runtime is specified as follows:
+
+```YAML
+generalOptions: [<comma>, <separated>, <options>]
+image: <registry>/<image name>:<version>
+commandOptions: [<comma>, <separated>, <options>]
+commandArgs: [<comma>, <separated>, <arguments>]
+```
+
+###### PodmanKubeRuntimeConfig
+
+The runtime configuration for the podman-kube runtime is specified as follows:
+
+```YAML
+generalOptions: [<comma>, <separated>, <options>]
+play_options: [<comma>, <separated>, <options>]
+down_options: [<comma>, <separated>, <options>]
+manifest: <string containing the K8s manifest>
+```
+
+#### CLI supports Ankaios manifest
+`swdd~cli-supports-ankaios-manifest~1`
+
+Status: approved
+
+The Ankaios CLI shall support the Ankaios manifest file format.
+
+Tags:
+- AnkaiosManifest
+
+Needs:
+- swdd
+- impl
+- utest
+- stest
+
+#### CLI provides a function to apply the given manifest file
+`swdd~cli-provides-apply-ankaios-manifest~1`
+
+Status: approved
+
+**When** the user provides a manifest file via the CLI command `ank apply manifest.yaml`
+
+**Then** the Ankaios CLI shall generate a state object for an update state request from the given manifest file
+
+**And** build a filter mask containing all workloads specified in the given manifest file
+
+**And** send an update state request to the Ankaios server containing the built state object and filter mask.
+
+Tags:
+- ApplyManifest
+
+Needs:
+- swdd
+- impl
+- utest
+- stest
+
+#### CLI provides a function to apply multiple manifest files
+`swdd~cli-provides-apply-multiple-ankaios-manifests~1`
+
+Status: approved
+
+**When** the user provides multiple manifest files via the CLI command `ank apply manifest1.yaml manifest2.yaml ...`
+
+**Then** the Ankaios CLI shall generate a state object for an update state request from the given manifest files
+
+**And** build a filter mask containing all workloads specified in the given manifest files
+
+**And** send an update state request to the Ankaios server containing the built state object and filter mask.
```

Comment:
We have a lot of repetitions in the actions and multiple requirements, how about having only one stating that we accept a list of files (and the list could have length 1) and another for the piping.

The same goes also for the -d option. We could just state that the `-d` means delete whatever the input was.